pandas/indexes/multi.py

```diff
@@ -656,9 +659,58 @@ def _has_complex_internals(self):
         return True
 
     @cache_readonly
+    def is_monotonic(self):
+
+        # TODO
+        # this is unfortunate we end up tupelizing
+        # just to determine monotonicity :<
+
+        # fast-path
+        if not self.levels[0].is_monotonic:
+            return False
+
```

Comment:
before 
```
In [1]: i = MultiIndex.from_product([np.arange(1000), np.arange(1000)], names=['one','two'])

In [2]: %time i.is_monotonic
CPU times: user 620 ms, sys: 42.1 ms, total: 662 ms
Wall time: 662 ms
```

```
In [2]: %time i.is_monotonic
CPU times: user 67.1 ms, sys: 14.1 ms, total: 81.2 ms
Wall time: 80.8 ms
```

@ssanderson thanks! this *seems* to pass, pushed up a commit with this. for sure an improvement.